lib/TTT/validity.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Validity do
+  def user_move_to_internal_state(move) do
+    try do
+      Map.get(
+        %{
+          1 => {0, 0},
+          2 => {0, 1},
+          3 => {0, 2},
+          4 => {1, 0},
+          5 => {1, 1},
+          6 => {1, 2},
+          7 => {2, 0},
+          8 => {2, 1},
+          9 => {2, 2}
+        },
+        String.to_integer(move),
+        {:error, :nonexistant_space}
+      )
+    rescue
+      ArgumentError -> {:error, :nonexistant_space}
+    end
+  end
+
+  def space_on_board?(move) do
+    cond do
+      user_move_to_internal_state(move) == {:error, :nonexistant_space} ->
+        {:error, :nonexistant_space}
+
+      true ->
+        {:ok, :space_on_board}
+    end
+  end
+
+  def number?(move) do
+    try do
+      String.to_integer(move)
```

Comment:
Same as above, this might be something you're only needing to do in here because you have an opportunity to sort of pre-shape `move` into an integer before `validity` gets it.

For example, what happens if move has to become more complex?  That may not happen in this project, so for the sake of a thought exercise, at least.  If `move` has to become like, a struct or something, or carry some additional information around (maybe we want to start tracking _when_ a move happens, as a random thought) - what changes need to happen to acommodate that, and where?

If we have a scenario where `move` is turned into the type of value we need it to be, most ideally at the time _when_ we get it, then that's straightforward: we just augment `move` itself, and maybe change how we're marshaling it.

If we have a scenario where `move` stays as a primitive value (in this case, a String) and everything reshapes itself so that it can work with Strings, then that sort of change is a lot more intensive in terms of the scope of work we need to do.  So, our code in that case would sort of resist that change.